src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs

```diff
@@ -14,13 +14,50 @@ namespace Microsoft.CodeAnalysis
     // all these are just helper methods
     internal partial class Checksum
     {
+        private static readonly Stack<IncrementalHash> s_hashStack = new Stack<IncrementalHash>();
```

Comment:
I think we can just use this.

http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/ObjectPool%25601.cs,40